piff/psf.py

```diff
@@ -116,4 +116,34 @@ def read(cls, file_name, logger=None):
                 logger.debug("interp = %s",interp)
         return cls(model,interp)
 
+    def generate_image(self, position, image, logger=None):
+        """Generates pixel array for star given position and galsim Image
 
+        :param position:    Position we are interpolating to.
+        :param image:       Galsim Image object on which we draw
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        :returns: a galsim image of the PSF
+        """
+
+        interpolated_params = self.interp.interpolate(position)
+        # set the parameters on the psf model, which gives a new model instance
+        interpolated_model = self.model.setParameters(interpolated_params)
+        # draw on the galsim image container
+        interpolated_image = interpolated_model.drawImage(image)
+        return interpolated_image
+
+    def generate_star(self, position, logger=None, **kwargs):
+        """Generates star given position and galsim Image
+
+        :param position:    Position we are interpolating to.
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        :returns: a star
+        """
+        from .stardata import StarData
+        # TODO: not sure what to do about image? for now just make a random galsim image
+        import galsim
+        image = galsim.Image(32, 32)
+        return StarData(self.generate_image(position, image, logger=logger),
+                        galsim.PositionD(*position), logger=logger, **kwargs)
```

Comment:
As mentioned below, `position` is not the right thing for `image_pos` here.  `position` is the location in the field of view where you want the nominal center of the PSF to be located.  StarData takes an `image_pos` which is the location in this postage stamp image where the nominal center is located.  Based on how you wrote `generate_image`, the right value here would be `image.trueCenter()`.

But see below for an alternate proposal for how this should work.